scalalib/src/mill/scalalib/UnidocModule.scala

```diff
@@ -0,0 +1,56 @@
+package mill.scalalib
+import mill._
+
+/**
+ * Mix this in to any [[ScalaModule]] to provide a [[unidocSite]] target that
+ * can be used to build a unified scaladoc site for this module and all of
+ * its transitive dependencies
+ */
+trait UnidocModule extends ScalaModule {
```

Comment:
This is new. We need it for the Mill scaladoc. Other folks probably need it too. It might not be perfect, but I figure let's get something in and people can improve it over time

Reply:
Very nice. I wonder whether we should put it in another package? I think, we should. The scalalib package contains already lots of various modules. Also, I think we eventually want to provide something like this for `JavaModule`s. A separate package makes grouping that stuff together easier.